openpdf/src/main/java/com/lowagie/text/HeaderFooter.java

```diff
@@ -209,6 +270,7 @@ public Paragraph paragraph() {
         if (before != null) {
             paragraph = new Paragraph(before.getLeading());
             paragraph.add(before);
+            paragraph.remove(paragraph.size()-1);
```

Comment:
This line is to delete line break "\n" in `paragraph`.
### For image
If I add image only, `before` will be a line break and footer will be like this:
![图片](https://user-images.githubusercontent.com/71996633/173038313-b4d51149-7130-4cf4-be57-75c254fe064f.png)
If there are texts below, it will become:
![图片](https://user-images.githubusercontent.com/71996633/173038890-cbda3662-46c0-4882-a176-8934d492ce40.png)

### For text
In addition, if there is an instance of `Paragraph` at the end of footer, a line break also appears, though it meets logic of program, I think there is no line break at the end of footer is better.
```java
Document document = new Document(PageSize.A4);
PdfWriter writer = PdfWriter.getInstance(document, new FileOutputStream("testText.pdf"));

Paragraph footerParagraph = new Paragraph();
String test = "This is a test line.";
String footerStr = "footer";

// testing section
footerParagraph.add(new Paragraph(footerStr));
// --------

HeaderFooter footer = new HeaderFooter(footerParagraph, false);
document.setFooter(footer);

document.open();
document.add(new Paragraph(test));
document.close();
```
![图片](https://user-images.githubusercontent.com/71996633/173039700-1cc5d608-d754-467e-a617-ed07b61d6b81.png)
**after removing "\n"**
![图片](https://user-images.githubusercontent.com/71996633/173040823-b7eed884-eba9-4dc3-a304-3b683fd3e1c5.png)

### compatibility
 And this line `paragraph.remove(paragraph.size()-1);` doesn't affect other line breaks
```java
// change codes in testing section above to followings
footerParagraph.add(new Paragraph(footerStr));
footerParagraph.add(new Paragraph(footerStr));
```
![图片](https://user-images.githubusercontent.com/71996633/173042038-d2089cf4-5e58-4d14-b8ed-e8c6fa4671a4.png)

Of course if you indeed want to add a line break at the end, you can add "\n".
```java
// change codes in testing section above to followings
footerParagraph.add(new Paragraph(footerStr));
footerParagraph.add("test\n");
footerParagraph.add(new Paragraph(footerStr));
footerParagraph.add("\n");
```
![图片](https://user-images.githubusercontent.com/71996633/173043555-31650a4f-abb2-416a-8e37-d0fecc55140b.png)

Reply:
this does break my headers and footers, it removes the only text in my header and footer